json-smart/src/main/java/net/minidev/json/parser/JSONParserInputStream.java

```diff
@@ -40,50 +37,55 @@ public JSONParserInputStream(int permissiveMode) {
 	public Object parse(InputStream in) throws ParseException {
 		return parse(in, ContainerFactory.FACTORY_SIMPLE, ContentHandlerDumy.HANDLER);
 	}
+	
+	/**
+	 * use to return Primitive Type, or String, Or JsonObject or JsonArray
+	 * generated by a ContainerFactory
+	 * 
+	 * @throws UnsupportedEncodingException If the named charset is not supported by an InputStreamReader
+	 */
+	public Object parse(InputStream in, String charset) throws ParseException, UnsupportedEncodingException {
+		return parse(in, ContainerFactory.FACTORY_SIMPLE, ContentHandlerDumy.HANDLER, charset);
+	}
 
 	/**
 	 * use to return Primitive Type, or String, Or JsonObject or JsonArray
 	 * generated by a ContainerFactory
+	 * 
+	 * NB: This uses the default charset
 	 */
 	public Object parse(InputStream in, ContainerFactory containerFactory) throws ParseException {
 		return parse(in, containerFactory, ContentHandlerDumy.HANDLER);
 	}
+	
+	/**
+	 * use to return Primitive Type, or String, Or JsonObject or JsonArray
+	 * generated by a ContainerFactory
+	 * 
+	 * @throws UnsupportedEncodingException If the named charset is not supported by an InputStreamReader
+	 */
+	public Object parse(InputStream in, ContainerFactory containerFactory, String charset) throws ParseException, UnsupportedEncodingException {
+		return parse(in, containerFactory, ContentHandlerDumy.HANDLER, charset);
+	}
 
 	/**
 	 * use to return Primitive Type, or String, Or JsonObject or JsonArray
 	 * generated by a ContainerFactory
+	 * 
+	 * NB: This uses the default charset
 	 */
 	public Object parse(InputStream in, ContainerFactory containerFactory, ContentHandler handler)
 			throws ParseException {
-		//
-		this.in = in;
-		this.pos = -1;
-		return super.parse(containerFactory, handler);
-	}
-
-	protected void read() throws IOException {
-		int i = in.read();
-		c = (i == -1) ? (char) EOI : (char) i;
-		pos++;
-		//
-	}
-
-	protected void readS() throws IOException {
-		sb.append(c);
-		int i = in.read();
-		if (i == -1) {
-			c = EOI;
-		} else {
-			c = (char) i;
-			pos++;
-		}
+	   return super.parse(new InputStreamReader(in));
```

Comment:
Should this use the params instead of ignoring what is passed in?
```suggestion
	   return super.parse(new InputStreamReader(in), containerFactory, handler);
```